src/framework/global/val.cpp

```diff
@@ -100,10 +100,8 @@ bool Val::toBool() const
         std::string str = std::get<std::string>(m_val);
         return str == VAL_TRUE;
     }
-    default:
-        return isNull() ? false : true;
     }
-    return false;
+    return isNull() ? false : true;
```

Comment:
This will cause a warning because after this change not all possible types will be covered in the switch statement

Reply:
I actually didn't get that warning. I was trying to fix an "unreachable" warning. I'll just delete the `return false;`

Reply:
It differs per compiler. MSVC tends to complain about unreachable code if you put the return statement in the `default` case and also put something at the bottom of the function, while Clang is afraid that the function doesn't always return a value if you don't. The general solution would be like this:
```c++
    default:
        break;
    }
    return isNull() ? false : true;
```